hhsuite/hhsearch.xml

```diff
@@ -1,43 +1,123 @@
+        </xml>
+    </macros>
     <requirements>
-        <requirement type="package" version="3.2.0">hhsuite</requirement>
+        <requirement type="package" version="@TOOL_VERSION@">hhsuite</requirement>
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
-        link '$hhm_ffdata' hhdb_hhm.ffdata &&
-        link '$hhm_ffindex' hhdb_hhm.ffindex &&
-        link '$cs219_ffdata' hhdb_cs219.ffdata &&
-        link '$cs219_ffindex' hhdb_cs219.ffindex &&
-        $method -e '$e' -i '$input' -d hhdb -o '$output'
-    ]]></command>
+        #if $db_source.db_source_selector == 'indexed':
+            ln -s '${db_source.ffindex.fields.path}.ffdata' hhdb_hhm.ffdata &&
+            ln -s '${db_source.ffindex.fields.path}.ffindex' hhdb_hhm.ffindex &&
+        #else
+            ln -s '$db_source.ffdata' hhdb_hhm.ffdata &&
+            ln -s '$db_source.ffindex' hhdb_hhm.ffindex &&
+        #end if
+
+        #if $db_source_cs219.db_source_selector == 'indexed':
+            ln -s '${db_source_cs219.ffindex.fields.path}.ffdata' hhdb_cs219.ffdata &&
+            ln -s '${db_source_cs219.ffindex.fields.path}.ffindex' hhdb_cs219.ffindex &&
+        #else
+            ln -s '$db_source_cs219.ffdata' hhdb_cs219.ffdata &&
+            ln -s '$db_source_cs219.ffindex' hhdb_cs219.ffindex &&
+        #end if
+
+        $method
+        -cpu \${GALAXY_SLOTS:-1}
+        -e '$e'
+        -i '$i'
+        -d hhdb
+        -o '$output'
+    ]]>    </command>
     <inputs>
-        <param format="data" name="input" type="data" label="Query Sequence" help="Single sequence or multiple sequence alignment (MSA)
-                in a3m, a2m, or FASTA format, or HMM in hhm format. (-i)"/>
-        <param format="txt" name="hhm_ffindex" type="data" label="HHM Index file" help="Database file ending with 'hhm.ffindex'."/>
-        <param format="data" name="hhm_ffdata" type="data" label="HHM Data file" help="Database file ending with 'hhm.ffdata'."/>
-        <param format="txt" name="cs219_ffindex" type="data" label="cs219 Index file" help="Database file ending with 'cs219.ffindex'."/>
-        <param format="data" name="cs219_ffdata" type="data" label="cs219 Data file" help="Database file ending with 'cs219.ffdata'."/>
+        <param argument="--i" type="data" format="fasta,hmm3" label="Query Sequence" help="Single sequence or multiple sequence alignment (MSA)
+                in FASTA format, or HMM in hhm3 format." />
         <param name="method" type="select" display="radio" label="Search Method" help="Select a search method. See help below for more information.">
             <option value="hhsearch" selected="true">HHsearch</option>
             <option value="hhblits">HHblits</option>
         </param>
-        <param name="e" type="float" label="E-value cutoff for inclusion in result alignment. (-e)" value="0.001" min="0" max="1"/>
+        <conditional name="db_source">
```

Comment:
Same as in the other tools, now you can optionally choose locally deployed indices.